launcher/internal/signaturediscovery/client.go

```diff
@@ -8,31 +8,37 @@ import (
 	"github.com/containerd/containerd"
 	"github.com/containerd/containerd/content"
 	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/remotes"
 	"github.com/google/go-tpm-tools/verifier/oci"
 	"github.com/google/go-tpm-tools/verifier/oci/cosign"
 	v1 "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 const signatureTagSuffix = "sig"
 
+type (
+	remoteResolverFetcher func(context.Context) (remotes.Resolver, error)
+	imageFetcher          func(context.Context, string, ...containerd.RemoteOpt) (containerd.Image, error)
+)
+
 // Fetcher discovers and fetches OCI signatures from the target repository.
 type Fetcher interface {
 	FetchImageSignatures(ctx context.Context, targetRepository string) ([]oci.Signature, error)
 }
 
 // Client is a wrapper of containerd.Client to interact with signed image manifest.
 type Client struct {
-	cdClient          *containerd.Client
 	OriginalImageDesc v1.Descriptor
-	RemoteOpts        []containerd.RemoteOpt
+	refreshResolver   remoteResolverFetcher
```

Comment:
`remoteResolverFetcher` or `refreshedResolverFetcher` to match the name below.

Reply:
Done.